File: BOJ(Java 공부용)/11650.java
import java.io.*;
import java.util.*;
public class Main {
    static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        p[] arr = new p[n];
        for (int i = 0; i < n; i++) arr[i] = new p(0, 0);
        for (int i = 0; i < n; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            int x = Integer.parseInt(st.nextToken()), y = Integer.parseInt(st.nextToken());
            arr[i].x = x;
            arr[i].y = y;
        }
        Arrays.sort(arr, (a, b) -> {
            if (a.x < b.x) return -1;
            else if (a.x > b.x) return 1;
            else {
                if (a.y < b.y) return -1;
                else if (a.y > b.y) return 1;
                else return 0;
            }
        });
        for (int i = 0; i < arr.length; i++) {
            bw.write(arr[i].x + " " + arr[i].y + "\n");
        }
        bw.flush();
        bw.close();
    }

    private static class p {
        int x, y;

        private p(int x, int y) {
            this.x = x;
            this.y = y;
        }
    }
}

File: BOJ(Kotlin)/11650.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.util.*
import kotlin.Comparator
import kotlin.collections.ArrayList

fun main(args: Array<String>) {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val bw = BufferedWriter(OutputStreamWriter(System.out))
    val n = br.readLine().toInt()
    val arr = mutableListOf<Pos>()
    for (i in 0 until n) {
        val st = StringTokenizer(br.readLine())
        val x = st.nextToken().toInt()
        val y = st.nextToken().toInt()
        arr.add(Pos(x,y))
    }

    arr.sortWith { o1, o2 ->
        when {
            o1.x < o2.x -> -1
            o1.x > o2.x -> 1
            else -> {
                when {
                    o1.y < o2.y -> -1
                    o1.y > o2.y -> 1
                    else -> 0
                }
            }
        }
    }

    for (value in arr) {
        bw.write(value.x.toString()+" "+value.y.toString()+"\n")
    }
    bw.flush()
    bw.close()
}
class Pos(val x : Int, val y: Int){}
